imageloading-core/src/main/java/com/google/accompanist/imageloading/AndroidDrawablePainter.kt

```diff
@@ -55,13 +57,30 @@ private val MAIN_HANDLER by lazy(LazyThreadSafetyMode.NONE) {
 class AndroidDrawablePainter(
     private val drawable: Drawable
 ) : Painter() {
-    private var invalidateTick by mutableStateOf(Unit, neverEqualPolicy())
+    private var invalidateTick by mutableStateOf(0)
+
+    private val callback: Drawable.Callback by lazy {
+        object : Drawable.Callback {
+            override fun invalidateDrawable(d: Drawable) {
+                // Update the tick so that we get re-drawn
+                invalidateTick++
+            }
+
+            override fun scheduleDrawable(d: Drawable, what: Runnable, time: Long) {
+                MAIN_HANDLER.postAtTime(what, time)
+            }
+
+            override fun unscheduleDrawable(d: Drawable, what: Runnable) {
+                MAIN_HANDLER.removeCallbacks(what)
+            }
+        }
+    }
 
     init {
         drawable.callback = object : Drawable.Callback {
```

Comment:
this one should be removed now? this will be overridden by another callback

Reply:
Woops, done